unit/java_bytecode/java_bytecode_parse_generics/parse_generic_class.cpp

```diff
@@ -16,307 +18,148 @@
+#include <iostream>
 
 SCENARIO(
-  "java_bytecode_parse_generics",
+  "java_bytecode_parse_generic_class_one_param",
   "[core][java_bytecode][java_bytecode_parse_generics]")
 {
-  const symbol_tablet &new_symbol_table=
-    load_java_class("generics", "./java_bytecode/java_bytecode_parse_generics");
+  const symbol_tablet &new_symbol_table = load_java_class(
+    "Generic",
+    "./java_bytecode/java_bytecode_parse_generics");
 
-  GIVEN("Some class files with Generics")
+  std::string class_prefix = "java::Generic";
+
+  WHEN("Parsing the class")
   {
-    WHEN("Parsing a class with type variable")
+    THEN("There is a generic class symbol Generic")
     {
-      REQUIRE(new_symbol_table.has_symbol("java::generics$element"));
-      THEN("The symbol type should be generic")
-      {
-        const symbolt &class_symbol=
-          new_symbol_table.lookup_ref("java::generics$element");
-        const typet &symbol_type=class_symbol.type;
+      REQUIRE(new_symbol_table.has_symbol(class_prefix));
 
-        REQUIRE(symbol_type.id()==ID_struct);
-        class_typet class_type=to_class_type(symbol_type);
-        REQUIRE(class_type.is_class());
-        java_class_typet java_class_type=to_java_class_type(class_type);
-        REQUIRE(is_java_generics_class_type(java_class_type));
-        java_generics_class_typet java_generics_class_type=
-          to_java_generics_class_type(java_class_type);
+      const symbolt &class_symbol = new_symbol_table.lookup_ref(class_prefix);
+      class_typet class_type =
+        require_symbol::require_complete_class(class_symbol);
+      java_generics_class_typet java_generics_class_type =
+        require_type::require_java_generic_class(class_type);
 
-        const struct_union_typet::componentt &elem=
-          java_generics_class_type.get_component("elem");
-        const typet &elem_type=java_class_type.component_type("elem");
-
-        REQUIRE(is_java_generic_parameter(elem_type));
+      THEN("It has type variable T")
+      {
+        REQUIRE(java_generics_class_type.generic_types().size() == 1);
+        const auto &generic_types = java_generics_class_type.generic_types();
 
-        REQUIRE(java_generics_class_type.generic_types().size()==1);
-        THEN("Type variable is named 'E'")
-        {
-          typet &type_var=java_generics_class_type.generic_types().front();
-          REQUIRE(is_java_generic_parameter(type_var));
-          java_generic_parametert generic_type_var=
-            to_java_generic_parameter(type_var);
-          REQUIRE(
-            generic_type_var.type_variable().get_identifier()==
-            "java::generics$element::E");
-          typet &sub_type=generic_type_var.subtype();
-          REQUIRE(sub_type.id()==ID_symbol);
-          symbol_typet &bound_type=to_symbol_type(sub_type);
-          REQUIRE(bound_type.get_identifier()=="java::java.lang.Object");
-        }
+        const java_generic_parametert &generic_param = generic_types[0];
```

Comment:
I might be tempted here to drop the `generic_types` variable and just use the compound expression entirely - e.g. `const java_generic_parametert &generic_param = java_generics_class_type.generic_types()[0]`